i18n-util/src/UrlUtil.js

```diff
@@ -1,49 +1,34 @@
-
+import { getLocale } from './LocaleUtil';
 
 const getUrlParameter = (name) => {
   const param = name.replace(/[\\[]/, '\\[').replace(/[\]]/, '\\]');
   var regex = new RegExp('[\\?&]' + param + '=([^&#]*)');
-  var results = regex.exec(window.location.hash);
+  var results = regex.exec(window.location.href);
   return (results && results.length > 1) ? decodeURIComponent(results[1].replace(/\+/g, ' ')) : null;
 };
 
-const getAliasUrl = (location) => {
-  let cshid = null;
-  if (location.hash && location.hash.length > 0) {
-    cshid = location.hash.split('#')[1];
-  }
-  var path = location.href.split('#')[0];
-  if (cshid) {
-    path += `#cshid=${cshid}`;
-  }
-  return path;
-};
-
-const getRedirectUrl = ({locale, aliasUrl}) =>  {
+const getRedirectUrl = () =>  {
+  // Extract locale from query
+  const locale = getLocale();
   // Add locale to url path and remove query param
   const fileName = 'okta_help.htm';
-  const host = aliasUrl.split('/okta_help.htm')[0];
+  const host = window.location.href.split('/okta_help.htm')[0];
   // Currently type is only used for end-user docs. In the future it will be used for admin docs
   const type = getUrlParameter('type');
-  let cshid = aliasUrl.split('okta_help.htm')[1];
-  // Remove locale form url
-  cshid = cshid.split('?locale=')[0];
-  // Remove type form url
-  cshid = cshid.split('?type')[0];
+  const id = getUrlParameter('id');
```

Comment:
@nikhilvenkatraman-okta just wondering if we have to validate this for any potential XSS attacks by using the query params. I wanted to run it by you just to be safe.

Reply:
@ganeshsomasundaram-okta 
`id` or any of the other parameters are not something that is submitted by a user from a form. All these parameters are coming from help links that are constructed within the monolith.